numpyro/hmc_util.py

```diff
@@ -769,9 +769,12 @@ def initialize_model(rng, model, *model_args, init_strategy=init_to_uniform, **m
                                     {k: v for k, v in constrained_values.items()},
                                     invert=True)
 
-    potential_fn = jax.partial(potential_energy, seeded_model, model_args, model_kwargs, inv_transforms)
+    # NB: we use model instead of seeded_model to prevent unexpected behaviours (if any)
+    potential_fn = jax.partial(potential_energy, model, model_args, model_kwargs, inv_transforms)
     if has_transformed_dist:
-        constrain_fun = jax.partial(constrain_fn, seeded_model, model_args, model_kwargs, inv_transforms)
+        # FIXME: why using seeded_model here triggers an error for funnel reparam example
+        # if we use MCMC class (mcmc function works fine)
+        constrain_fun = jax.partial(constrain_fn, model, model_args, model_kwargs, inv_transforms)
```

Comment:
I keep getting the error
```
Exception: Can't lift sublevels 2 to 1 
```
when using `seeded_model` with MCMC class in funnel reparam model.
```
  File "examples/funnel.py", line 46, in reparam_model                                                  
    y = numpyro.sample('y', dist.Normal(0, 3))                                                          
  File "/home/fehiepsi/numpyro/numpyro/primitives.py", line 62, in sample                               
    msg = apply_stack(initial_msg)                                                                      
  File "/home/fehiepsi/numpyro/numpyro/primitives.py", line 12, in apply_stack                          
    handler.process_message(msg)                                                                        
  File "/home/fehiepsi/numpyro/numpyro/handlers.py", line 334, in process_message                       
    self.rng, rng_sample = random.split(self.rng)                                                       
  File "/home/fehiepsi/miniconda3/envs/pydata/lib/python3.6/site-packages/jax/random.py", line 193, in s
plit                                                                                                    
    return _split(key, num)      
```
Anyway, I think it is safe to use `model` here because there should not have any stochastic behaviour at potential_fn and constrain_fn.

Because I was feeling that this is a dangerous issue so I mark this as `FIXME`. Will remove it when I understand the reason behind.

Reply:
I believe that using `model` instead of `seeded_model` also makes things faster (we don't have to split keys when evaluating `potential_fn`.

Reply:
While this change looks okay to me, I would like to better understand the failure when using `seeded_model` since this might be indicative of other underlying issues.